crate/kmip/src/crypto/cover_crypt/master_keys.rs

```diff
@@ -159,21 +164,92 @@ fn create_master_public_key_object(
     })
 }
 
+/// Update a Covercrypt policy associated to a master key.
+pub fn update_policy(policy: &mut Policy, action: &RekeyEditAction) -> Result<(), KmipError> {
+    match action {
+        RekeyAccessPolicy(_) | PruneAccessPolicy(_) => Ok(()),
+        RemoveAttribute(attrs) => attrs
+            .iter()
+            .try_for_each(|attr| policy.remove_attribute(attr)), // TODO: tests revoking of existing keys with deleted attribute?
+        DisableAttribute(attrs) => attrs
+            .iter()
+            .try_for_each(|attr| policy.disable_attribute(attr)),
+        RenameAttribute(pairs_attr_name) => pairs_attr_name
+            .iter()
+            .try_for_each(|(attr, new_name)| policy.rename_attribute(attr, new_name.clone())),
+        AddAttribute(attrs_properties) => {
+            attrs_properties
+                .iter()
+                .try_for_each(|(attr, encryption_hint)| {
+                    policy.add_attribute(attr.clone(), *encryption_hint)
+                })
+        }
+    }
+    .map_err(|e| {
+        KmipError::KmipError(
+            ErrorReason::Unsupported_Cryptographic_Parameters,
+            e.to_string(),
+        )
+    })?;
+
+    Ok(())
+}
+
 /// Update the master key with a new Policy
-/// (after rotation of some attributes typically)
+/// (after editing the policy typically)
 pub fn update_master_keys(
     cover_crypt: &Covercrypt,
     policy: &Policy,
+    action: &RekeyEditAction,
     master_private_key: &Object,
     master_private_key_uid: &str,
     master_public_key: &Object,
     master_public_key_uid: &str,
 ) -> Result<(Object, Object), KmipError> {
+    let (mut msk, mut mpk) =
+        covercrypt_keys_from_kmip_objects(master_private_key, master_public_key)?;
```

Comment:
As modified in the merged PR, type alias is used to associated UIDs with their associated `Object`. The UID is not contained in these objects, it is thus not possible to avoid passing these UIDs around without structural modifications of the `Object` type.